lib/overcommit/hook_loader/base.rb

```diff
@@ -22,6 +22,34 @@ def load_hooks
 
     private
 
+    # GNU/Emacs-style error format:
+    AD_HOC_HOOK_DEFAULT_MESSAGE_PATTERN =
+      /^(?<file>(?:\w:)?[^:]+):(?<line>\d+):[^ ]* (?<type>[^ ]+)/.freeze
```

Comment:
Have we thought about not providing a default?

There are many potential hooks where all that matters is a pass/fail. If we can piggyback on the existing support for ad hoc hooks to simply run a specified script, and rather than try to offer a default we simply ensure that the user provides a `message_pattern` regex. This would likely reduce confusion since it forces the user to explicitly define the format rather than expect it to Just Work™.

Reply:
I am not sure if by piggybacking you mean I should merge the code of both kinds of ad-hoc hooks? I'd rather keep it as it is now, else RuboCop will rightfully complain about too high metrics!

About not having a default for `message_pattern`: I empathize with your "too much magic" concern, although I am quite fond of this versatile default regex (not my baby, I think I stole it from some existing hook). Maybe we could keep it only as a good example/template in the docs.
If your only concern is having some option to detect message extraction, maybe we could use another one instead: maybe (as initially envisioned in #750) we could introduce `extract_messages_from` with a mandatory value (only `stdout` for now, but `stderr` would be trivial to add) and use that?

Reply:
>Maybe we could keep it only as a good example/template in the docs.
Totally fine with that!

I hadn't considered the need to potentially decide between `stdout`/`stderr`. Slight preference in this case would be to create a wrapper script that sets up the redirects appropriately, rather than requiring Overcommit to address this issue. For example:

```yaml
# .overcommit.yml
PreCommit:
  CustomScript:
    enabled: true
    required_executable: './bin/custom-script.sh'
```

```bash
# custom-script.sh
#!/bin/bash

bundle exec rubocop 2>&1
```

Thoughts on leaving out `extract_messages_from` given the above is an option?

Reply:
I thought needing wrappers would be a non-Unix-portability issue for the project (only Windows without WSL or such nowadays I guess... and there's always PowerShell), but that's fine with me, I can live with wrappers and Windows being at a disadvantage! ;-)

I'll address that in the next iteration.

Reply:
As requested, `extract_messages_from` is now gone, and `message_pattern` no longer has a "magic" default value.